here-naksha-lib-psql/src/jvmMain/resources/naksha.sql

```diff
@@ -243,13 +243,13 @@ BEGIN
     encoding = encoding & 14;
   end if;
   if (encoding = 0) then
-    RETURN ST_GeomFromTWKB(geo);
+    RETURN ST_SetSRID(ST_GeomFromTWKB(geo), 4326); -- TWKB does not encode SRID, we force the common one
   elsif (encoding = 2) then
     RETURN ST_GeomFromWKB(geo);
   elsif (encoding = 4) then
     RETURN ST_GeomFromEWKB(geo);
   elsif (encoding = 6) then
-    RETURN ST_GeomFromGeoJSON(geo::text);
+    RETURN ST_GeomFromGeoJSON(convert_from(geo, 'UTF8'));
```

Comment:
We had a bug here, casting `bytea::text` won't work as expected, I was getting `SQL Error [XX000]: ERROR: unexpected character (at offset 0)`. The correct way to get text represntation of JSON previously encoded as byte array is to use `convert_from` with `UTF8` encoding specified (this is also the encoding we use in lib-psql).

Some snippets to prove it step by step:
1) Hex representation of simple Line String:
```
// Kotlin:
LineStringCoord(
    PointCoord(longitude = 25.0, latitude = 25.0),
    PointCoord(longitude = 25.0, latitude = 26.0),
)

// bytes (hex):
'\x7b2274797065223a224c696e65537472696e67222c22636f6f7264696e61746573223a5b5b32352e302c32352e305d2c5b32352e302c32362e305d5d7d'
```

2) This is what happened before - confirming that we passed something invalid to `ST_GeomFromGeoJSON`: 
```
> SELECT ST_GeomFromGeoJSON((bytea '\x7b2274797065223a224c696e65537472696e67222c22636f6f7264696e61746573223a5b5b32352e302c32352e305d2c5b32352e302c32362e305d5d7d')::text)

SQL Error [XX000]: ERROR: unexpected character (at offset 0)
```

3) Checking the cast - we can see that we did not get json, rather raw bytes as text
```
SELECT (bytea '\x7b2274797065223a224c696e65537472696e67222c22636f6f7264696e61746573223a5b5b32352e302c32352e305d2c5b32352e302c32362e305d5d7d')::text

\x7b2274797065223a224c696e65537472696e67222c22636f6f7264696e61746573223a5b5b32352e302c32352e305d2c5b32352e302c32362e305d5d7d
```

4) Passing these byte representation is not suitable for `ST_GeomFromGeoJSON` - we get the same error as above
```
SELECT ST_GeomFromGeoJSON('\x7b2274797065223a224c696e65537472696e67222c22636f6f7264696e61746573223a5b5b32352e302c32352e305d2c5b32352e302c32362e305d5d7d')

SQL Error [XX000]: ERROR: unexpected character (at offset 0)
```

5) Switching from cast to `convert_from`, we get proper JSON
```
SELECT convert_from(bytea '\x7b2274797065223a224c696e65537472696e67222c22636f6f7264696e61746573223a5b5b32352e302c32352e305d2c5b32352e302c32362e305d5d7d', 'UTF8')

{"type":"LineString","coordinates":[[25.0,25.0],[25.0,26.0]]}
```

6) Passing proper JSON to `ST_GeomFromGeoJSON` we get what wanted - `geometry` instance
```
SELECT ST_GeomFromGeoJSON(convert_from(bytea '\x7b2274797065223a224c696e65537472696e67222c22636f6f7264696e61746573223a5b5b32352e302c32352e305d2c5b32352e302c32362e305d5d7d', 'UTF8'))

LINESTRING (25 25, 25 26)
```

Reply:
Agree, it seems that `::text` casting uses the default database encoding, which is "C", therefore it does not recognize that the string is `UTF8`.